main.go

```diff
@@ -136,129 +140,38 @@ func updateHandler(w http.ResponseWriter, req *http.Request) {
 
 // Main index page handler.
 func mainHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		http.Error(w, "Method should be GET.", http.StatusMethodNotAllowed)
+		return
+	}
+
 	if err := loadTemplates(); err != nil {
 		fmt.Fprintln(w, "loadTemplates:", err)
 		return
 	}
 
-	//started := time.Now()
-
-	CommonHat(w)
-	defer CommonTail(w)
+	w.Header().Set("Content-Type", "text/html; charset=UTF-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
 
-	io.WriteString(w, `<div id="checking_updates"><h2 style="text-align: center;">Checking for updates...</h2></div>`)
-	io.WriteString(w, `<div id="no_updates" style="display: none;"><h2 style="text-align: center;">No Updates Available</h2></div>`)
-	defer io.WriteString(w, `<script>document.getElementById("checking_updates").style.display = "none";</script>`)
+	_ = commonHead(w)
+	defer func() { _ = commonTail(w) }()
```

Comment:
I should handle the error, but honestly, I'm not sure how to handle an error writing to an `http.ResponseWriter`, aside from maybe calling `log.Println(err)`. Doing `http.Error` is not viable since by then the headers might have already been written (successfully).